src/main/scala/dpla/ingestion3/mappers/providers/NcMapping.scala

```diff
@@ -0,0 +1,190 @@
+    extractStrings(data \\ "mods" \ "physicalDescription" \ "form")
+
+  override def identifier(data: Document[NodeSeq]): Seq[String] =
+  // <mods:identifier>
+    extractStrings(data \\ "mods" \ "identifier")
+
+  override def language(data: Document[NodeSeq]): Seq[SkosConcept] =
+  // <mods:language><mods:languageTerm>
+    extractStrings(data \\ "mods" \ "language" \ "languageTerm")
+      .map(nameOnlyConcept)
+
+  override def place(data: Document[NodeSeq]): Seq[DplaPlace] =
+  // <mods:subject><mods:geographic>
+    extractStrings(data \\ "mods" \ "subject" \ "geographic")
+      .map(nameOnlyPlace)
+
+  override def publisher(data: Document[NodeSeq]): Seq[EdmAgent] =
+  // <mods:originInfo><mods:publisher>
+    extractStrings(data \\ "mods" \ "originInfo" \ "publisher")
+      .map(nameOnlyAgent)
+
+  override def relation(data: Document[NodeSeq]): ZeroToMany[LiteralOrUri] =
+  // <relatedItem><location><url> and/or <relatedItem><titleInfo><title>
+    (extractStrings(data \\ "mods" \ "relatedItem" \ "location" \ "url") ++
+      extractStrings(data \\ "mods" \ "relatedItem" \ "titleInfo" \ "title"))
+      .map(eitherStringOrUri)
+
+  override def rights(data: Document[NodeSeq]): AtLeastOne[String] =
+  // <accessCondition type="local rights statements">
+    (data \\ "mods" \ "accessCondition")
+      .filter(node => filterAttribute(node, "type", "local rights statements"))
+      .flatMap(extractStrings)
+
+  override def subject(data: Document[NodeSeq]): Seq[SkosConcept] =
+    // <mods:subject><mods:topic>
+    extractStrings(data \\ "mods" \ "subject" \ "topic").map(nameOnlyConcept)
+
+  override def title(data: Document[NodeSeq]): Seq[String] =
+  // <mods:titleInfo><mods:title>
+    extractStrings(data \\ "mods" \ "titleInfo" \ "title")
+
+  override def `type`(data: Document[NodeSeq]): Seq[String] =
+  // <mods:genre>
+    extractStrings(data \\ "mods" \ "genre")
+
+  // OreAggregation
+  override def dplaUri(data: Document[NodeSeq]): ZeroToOne[URI] = mintDplaItemUri(data)
+
+  override def dataProvider(data: Document[NodeSeq]): ZeroToMany[EdmAgent] =
+    // first <note type="ownership">
+    (data \\ "mods" \ "note")
+      .filter(node => filterAttribute(node, "type", "ownership"))
+      .flatMap(extractStrings)
+      .map(nameOnlyAgent)
+      .take(1)
+
+  override def edmRights(data: Document[NodeSeq]): ZeroToMany[URI] =
+  // <accessCondition type="use and reproduction">
+    (data \\ "mods" \ "accessCondition")
+      .filter(node => filterAttribute(node, "type", "use and reproduction"))
+      .flatMap(extractStrings)
+      .map(URI)
+
+  override def intermediateProvider(data: Document[NodeSeq]): ZeroToOne[EdmAgent] = {
+    // second <note type=ownership> if it exists
+    val providers = (data \\ "mods" \ "note")
+      .filter(node => filterAttribute(node, "type", "ownership"))
+      .flatMap(extractStrings)
+      .map(nameOnlyAgent)
+
+    if (providers.length > 1)
+      Some(providers(1))
+    else None
```

Comment:
I think you could do `.lift(1)` instead of the `if-else` here.  It will return an `Option` with `None` if there is no element at index 1.

Reply:
👍 thanks for the reminder